include/AST/Types/FunctionTy.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef GLU_AST_TYPES_FUNCTIONTY_HPP
+#define GLU_AST_TYPES_FUNCTIONTY_HPP
+
+#include "TypeBase.hpp"
+#include <vector>
+
+namespace glu::types {
+
+/// @brief FunctionTy is a class that represents a function type in the AST.
+class FunctionTy : public TypeBase {
+    std::vector<TypeBase *> const _parameters;
+    TypeBase *_returnType;
```

Comment:
if params are const why not return type?
```suggestion
    TypeBase *const _returnType;
```